### ISSUE TITLE
[bug] Denon AVR Source Select dropdown cleared on save

### ISSUE THREAD

**Issue by geransmith** [unfoldedcircle/bug-reports#104]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

With the Denon AVR-X3700H configured inside Home Assistant, when trying to configure macros or activities, if I add the "Input Source" command and select a source in the dropdown, the value I choose is not persisted after leaving the page. This leads to an error executing the macro or activity.

### How to Reproduce

Prerequisites: Have Home Assistant configured with the Denon addon and multiple sources on the AVR setup.

1. In the Web Configurator, select the option to add a Macro
2. Add the Denon Receiver entity from Home Assistant
3. Select the command "Input Source"
4. Choose a source in the dropdown
5. Back out of the macro using the UIs back arrow
6. Attempt to run the Macro

### Expected behavior

The source should persist and be saved.

### System version

1.3.3

### What part of the system affected by the problem?

Integration

**Comment by splattner** [comment]
I think its not only with DenonAVR/homeassistant but all Select Source Commands? Have the same issue with my own implemented media player for denon

When I look on the payload made by the web-configurator I don't see the `source` parameter:

```
{
  "name": {
    "de": "Watch TV",
    "en": "Watch TV"
  },
  "icon": "uc:activity",
  "description": {},
  "options": {
    "entity_ids": [
      "denonavr.main.mediaplayer",
      "uc.main.9948bdad-ab48-4908-a118-2b154e6113f9",
      "uc_androidtv_driver.main.D4F8292E8A2C"
    ],
    "sequences": {
      "on": [
        {
          "command": {
            "cmd_id": "POWER_ON",
            "entity_id": "uc.main.9948bdad-ab48-4908-a118-2b154e6113f9"
          },
          "type": "command"
        },
        {
          "command": {
            "cmd_id": "media_player.on",
            "entity_id": "denonavr.main.mediaplayer",
            "params": {}
          },
          "type": "command"
        },
        {
          "command": {
            "cmd_id": "media_player.on",
            "entity_id": "uc_androidtv_driver.main.D4F8292E8A2C",
            "params": {}
          },
          "type": "command"
        },
        {
          "command": {
            "cmd_id": "media_player.select_source",
            "entity_id": "denonavr.main.mediaplayer",
            "params": {}
          },
          "type": "command"
        }
      ],
      "off": [
        {
          "command": {
            "cmd_id": "POWER_OFF",
            "entity_id": "uc.main.9948bdad-ab48-4908-a118-2b154e6113f9"
          },
          "type": "command"
        },
        {
          "command": {
            "cmd_id": "media_player.off",
            "entity_id": "denonavr.main.mediaplayer",
            "params": {}
          },
          "type": "command"
        },
        {
          "command": {
            "cmd_id": "media_player.off",
            "entity_id": "uc_androidtv_driver.main.D4F8292E8A2C",
            "params": {}
          },
          "type": "command"
        }
      ]
    },
    "prevent_sleep": true
  }
}
```

**Comment by Tim-GitHub** [comment]
I am seeing this same behavior with my Onkyo AVR and my LG TV, which are both configured in Home Assistant (in my case the command is labled "Input source" rather than "Select source" but I assume they are the same thing). Adding via Activity or Macro the input is not saved. If I save it and close the Activity (hoping it is saved) and attempt to run the Activity via the Remote Two, I get a red error "BadRequest Invalid or missing  params.source attri...". Also when adding a button for the Activity UI, while is does save the input name, it does not seem to work either. (There is no drop down, I had to free type the input name).

Also, maybe consider renaming this issue to "Select source dropdown cleared on save" since it seems to impact more than just Denon?

**Comment by zehnm** [comment]
I'm on it. Got the same equipment, should be easy to reproduce and fix it.

**Comment by Seexelas** [comment]
I have the same issue, with my Zidoo UHD5k, Integra AVR 5.4, Apple Tv4k Gen3 and Panasonic GZ2000. The Apple Tv is in integration and the rest via Home Assistant. 

What is weird is that existing Activity can be modified without issue, only the new or duplicate one.

**Comment by zehnm** [comment]
Confirmed: the web-configurator doesn't set the selected item into the `command.params.source` field when updating the activity after selection.

It's a generic error for all media player entities with input source feature.
I've also noticed, that the dropdown selection is only shown in the activity sequence and not in the UI editor. Here it's a free text input field:

<img width="429" alt="image" src="https://github.com/unfoldedcircle/feature-and-bug-tracker/assets/17356629/1e980bb5-0fea-4822-acb2-7359d74a8a23">

I'll add this to our backlog as well.
